shared/src/reentrancy_guard.rs

```
@@ -0,0 +1,47 @@
+    ReentrantCall = 6001,
+}
+
+/// Storage key for re-entrancy guard status
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ReentrancyGuardKey {
+    pub guard: Symbol,
+}
+
+const REENTRANCY_GUARD_KEY: ReentrancyGuardKey = ReentrancyGuardKey {
+    guard: symbol_short!("REENTRANCY_GUARD"),
```

Clippy (1.95.0)

[error] symbol too long

(error)